features/networking/pod.feature

```diff
@@ -190,3 +190,82 @@ Feature: Pod related networking scenarios
     Then the step should succeed
     And the output should not contain "ingress_policing_rate: 1953"
 
+  # @author anusaxen@redhat.com
+  # @case_id OCP-19810
+  @admin
+  Scenario: Conntrack rule for UDP traffic should be removed when the pod for NodePort service deleted
+   Given I store the schedulable nodes in the :nodes clipboard
+   Given I use the "<%= cb.nodes[0].name %>" node
+   And I have a project
+   
+   When I run the :create client command with:
+     | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/pod_with_udp_port_listener.json |
+   Then the step should succeed
+   Given a pod becomes ready with labels:
+     | name=udp-pods |
+   And evaluation of `pod.name` is stored in the :host_pod1_name clipboard
+   And evaluation of `pod.ip` is stored in the :host_pod1_ip clipboard
+   And evaluation of `host_subnet(cb.nodes[0].name).ip` is stored in the :node_ip clipboard
+   
+   When I run the :expose client command with:
+     | resource          | pod |
+     | resource_name     | <%= cb.host_pod1_name %> |
+     | type              | NodePort |
+     | port              | 8080 |
+     | protocol          | UDP |
+   Then the step should succeed
+   
+   When I run the :get client command with:
+     | resource      | service |
+     | output        | jsonpath='{.items[*].spec.ports[*].nodePort}' | 
+   Then the step should succeed
+   And evaluation of `@result[:response]` is stored in the :nodeport clipboard
+   
+   When I run the :create client command with:
+     | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/aosqe-pod-for-ping.json |
+   Then the step should succeed
+   Given a pod becomes ready with labels:
+     | name=hello-pod |
+   And evaluation of `pod.name` is stored in the :client_pod clipboard
+   And evaluation of `host_subnet(cb.nodes[1].name).ip` is stored in the :client_ip clipboard
+   
+   # 'yes' command will send a character "h" continously for 3 seconds to /dev/udp on listener where the node is listening for udp traffic on exposed nodeport. The 3 seconds mechanism will create an Assured
+   #  entry which will give us enough time to validate upcoming steps 
+   When I run the :exec background client command with:
+     | pod              | <%= cb.client_pod %> |
+     | oc_opts_end      |    |
+     | exec_command     | bash |
+     | exec_command_arg | -c |
+     | exec_command_arg | yes "h">/dev/udp/<%= cb.node_ip %>/<%= cb.nodeport %> |
```

Comment:
I don't think `bash` supports making connections in Red Hat linux.

In any case, as Peter said, this will not work with internal IPs. You can probably test one node by running some command on another node.

Additionally relying on local shell commands is veeery discouraged. Either use pure ruby code, run commands in pods, or ocmmands on nodes. In that order of preference.

Reply:
I agree Alex. For 4.0, this would need some tweak.
Regarding Bash support in RHEL, i am executing this command on pod. 
So when i did this:
oc exec pod_name -- sh -c yes "h">/dev/udp/node_ip/udp_port
**error**: cant find /dev/udp/node_ip/udp_port
And when i did "bash" above it took it. I guess the reusable "hello-pod" docker image might have built on "FROM docker.io/centos:7"" version which supports sh but conntrack entries showed that connections were honored at the listener side (RHEL) using bash

The pod, as per my experiment didn't support "sh" but "bash". I was in favor of "sh" though. I am not sure if sh is favoured then in that case i can create a dockerfile/image using "FROM docker.io/centos:7"

Reply:
reusable "hello-pod" docker image might **NOT** have

Reply:
Yeah, the aos-pod-for-ping image is built on top of alpine, which may not be functional as centos.
But as the bash is executed inside the container and we can get the expected result.
I am not sure what is the problem though.

Reply:
Since this test won't work for OCP >= 4.0, perhaps we can add a statement to fail early in the OCP version does not meet the criteria

Reply:
Sounds good to me, Peter. Testing is going on in a parallel though on 4.0 but no luck yet. Investigation is on..

Reply:
@anuragthehatter, ok, that's why I suggested to gate the testcase by checking on the ocp version of the current master.  You can also change Polarion to specify that only OCP <= 3.11 applies to this test case.

Reply:
Thanks for Info,Peter. I will discuss with @bmeng today eve and will conclude the appropriate step

Reply:
@pruan-rht I am going to gate the test case by checking 'oc version' on master as discussed with @bmeng. 'oc version' doesn't support json output on 3.xx so i am going to try grep/awk to fetch that value and the abort if value doesn't match 3.xx. Thanks

Reply:
Master branch now only supports 4.0+. So you can update things to work with 4.0. the `v3` branch is for 3.x